### DISCUSSION TITLE
Is "Support for Lucy DSL (domain-specific language)" still planed?

### DISCUSSION THREAD

**Discussion by PabloRuizCuevas** [statelyai/xstate#4101] [Q&A] [answered]
In the page of the docs https://xstate.js.org/docs/visualizer/#keyboard-shortcuts-and-command-palette it says that support for Lucy DSL is in the pipeline, but i don't see any other reference in xtsate docs or github to it.

I find the syntax of [lucy]( https://lucylang.org/ ) extremely convenient, but i guess the project seems not very active, what is the plan with this regard, is the planed feature abandoned?

**Comment by davidkpiano** [comment, marked as answer]
We would like to support Lucy and other custom DSLs, but as you said, Lucy may be out-of-date currently.

There is an in-progress project called https://www.stateml.org/ that is very similar and I think still active, that may also be worth pursuing.

**Reply by ChrisShank** [reply]
thanks for the shout out :) Still active and proving out parts of the language. I cant give any specific date when we open source the language, but progress has been slow and steady.

**Reply by PabloRuizCuevas** [reply]
Good to know, thanks! a DSL in this matter is very much appreciated :)
